### DISCUSSION TITLE
API: Breakdown custom events by prop and get (non unique) visitors

### DISCUSSION THREAD

**Discussion by functino** [plausible/analytics#1662] [Anything Else]
In the documentation there is an example how to breakdown an event by prop:

https://plausible.io/docs/stats-api#breakdown-custom-event-by-custom-props

So this request:
```
curl https://plausible.io/api/v1/stats/breakdown?site_id=$SITE_ID&period=6mo&property=event:props:method&filters=event:name%3D%3DDownload
  -H "Authorization: Bearer ${TOKEN}"
```

Will give you a result like this:
```{
    "results": [
        {
            "method": "HTTP",
            "visitors": 1477
        },
        {
            "method": "Magnet",
            "visitors": 370
        }
    ]
}
```

But the number of `visitors` here are `unique` events. Is there a way to get the number of total events via the API?

**Comment by RobertJoonas** [comment]
Hi @functino! To also get the number of total conversions with this query, you can add `metrics=visitors,events` to it. Or `metrics=events` if you're only interested in total conversions.

**Reply by functino** [reply]
Works like a charm. Thank you very much. I somehow missed `events` in the list of allowed `metrics` values.

**Reply by mbucak** [reply]
can you share you query? I am getting this error:
```
{
    "error": "The metric `events` is not recognized. Find valid metrics from the documentation: https://plausible.io/docs/stats-api#get-apiv1statsbreakdown"
}
```

when I use this query:
`http://localhost:8000/api/v1/stats/breakdown?site_id=10.33.11.173:5560/index.html&property=event:props:videoId&filters=event:name%3D%3Dplay&metrics=events`

**Reply by functino** [reply]
@mbucak this is the url I used in the end:
`https://plausible.io/api/v1/stats/breakdown?site_id=${siteId}&metrics=events&period=${period}&property=event:props:value&filters=event:name%3D%3D${event}`

**Reply by leewan-09** [reply]
> can you share you query? I am getting this error:
> 
> ```
> {
>     "error": "The metric `events` is not recognized. Find valid metrics from the documentation: https://plausible.io/docs/stats-api#get-apiv1statsbreakdown"
> }
> ```
> 
> when I use this query: `http://localhost:8000/api/v1/stats/breakdown?site_id=10.33.11.173:5560/index.html&property=event:props:videoId&filters=event:name%3D%3Dplay&metrics=events`

Were you able to solve this problem? I'm having the same issue.

**Reply by RobertJoonas** [reply]
The `events` metric is not yet supported in self-hosted Plausible, unfortunately.

Our self-hosted version is not updated as frequently as the Plausible managed hosting in the cloud, which changes daily. This means that the latest features won't be immediately available. think of self-hosted as a long-term release. we stack all the new features and fixes that have been battle-tested on the hosted version and release a new self-hosted version twice a year or so. We don't have any more specific timelines than this as we have limited resources, and the self-hosted release depends on the free capacity of our developers.

Thanks for understanding!
